common/deftype/deftype.cpp

```diff
@@ -47,31 +47,39 @@ static IAtom * emptyAtom;
 static CriticalSection * typeCS;
 static TypeCache * globalTypeCache;
 
-static CBoolTypeInfo *btt = NULL;
-static CBlobTypeInfo *bltt = NULL;
-static CVoidTypeInfo *vtt = NULL;
-static CNullTypeInfo *ntt = NULL;
-static CRecordTypeInfo *rtt = NULL;
-static CAnyTypeInfo *anytt = NULL;
-static CPatternTypeInfo *patt = NULL;
-static CTokenTypeInfo *tokentt = NULL;
-static CFeatureTypeInfo *featurett = NULL;
+static std::atomic<CBoolTypeInfo*> btt(nullptr);
+static std::atomic<CBlobTypeInfo*> bltt(nullptr);
+static std::atomic<CVoidTypeInfo*> vtt(nullptr);
+static std::atomic<CNullTypeInfo*> ntt(nullptr);
+static std::atomic<CRecordTypeInfo*> rtt(nullptr);
+static std::atomic<CPatternTypeInfo*> patt(nullptr);
+static std::atomic<CTokenTypeInfo*> tokentt(nullptr);
+static std::atomic<CFeatureTypeInfo*> featurett(nullptr);
+static std::atomic<CEventTypeInfo*> ett(nullptr);
+static std::atomic<CAnyTypeInfo*> anytt(nullptr);
+
 static CStringTypeToTypeMap * stt;
 static CStringTypeToTypeMap * datatt;
 static CStringTypeToTypeMap * vstt;
 static CStringTypeToTypeMap * qstt;
 static CUnicodeTypeToTypeMap * utt;
 static CUnicodeTypeToTypeMap * vutt;
 static CUnicodeTypeToTypeMap * u8tt;
-static CIntTypeInfo *itt[2][8];
-static CSwapIntTypeInfo *sitt[2][8];
+
+// Note: C++ standard does not guarantee std::atomic will initialize memory
```

Comment:
This comment is not true for static variables - they are default zero-initialised.  The initialiser list is not required.